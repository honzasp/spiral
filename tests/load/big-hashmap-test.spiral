(program
  (import std)
  (import std.hashmap)
  (import std.hash)
  (import std.test)

  (test "big map of integers added in order" (lambda (t)
    (let ((limit 100000)
          (h (hashmap-new hash-int ==)))
      (do ((i 0 (+ i 1)))
          ((>= i limit))
        (hashmap-set! h i (* 42 i)))
      (do ((i 0 (+ i 1)))
          ((>= i limit))
        (let ((k (mod (* i 123) 53)))
          (assert-eqv t (hashmap-get h k) (* 42 k)))))))
          
  (test "big map of integers added out of order" (lambda (t)
    (let ((prime 1299817)
          (h (hashmap-new hash-int ==)))
      (do ((i 0 (+ i 1)))
          ((>= i prime))
        (hashmap-set! h (mod (* 123 i) prime) i))
      (do ((i 0 (+ i 1)))
          ((>= i prime))
        (let ((k (mod (* 123 i) prime)))
          (assert-eqv t (hashmap-get h k) i)))))))
