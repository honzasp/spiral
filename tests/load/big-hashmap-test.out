test big map of integers added in order ok
test big map of integers added out of order ok
